using-directus-as-a-baby-health-tracker/index.md

```diff
@@ -0,0 +1,403 @@
+### Flows
+
+> [!TIP] What is a Flow
+> Flows enable custom, event-driven data processing and task automation within Directus.  [Learn more about Flows](https://docs.directus.io/reference/system/flows.html) 
+
+#### Read Data Flow
+The by far most complex Flow of this use case is the one we need to extract the data. This is due to the lack of publicly documented API of the used health-tracking gadget. 
+As there are a couple of Python libraries fetching the data, it was not the easiest way, but with a basic understanding of Python and programming in general, it was possible to kind of reverse-engineer this code and translate each step into an Operation for the Flow.
+The very rough workflow looks like this:
+1. Define the trigger
+2. Authenticate and get all the needed tokens
+3. Get a list of connected devices
+4. Get the data of our target device
+5. Transform the data to our needs
+6. Save the data
+##### The Trigger
+The easiest way to get regular data fetched automatically with a Flow is a scheduled trigger.
+During the setup, we've set the trigger to the following settings:
+
+Status: Active
+Activity & Log tracking: Track Log & Activity 
+Interval: * * * * * (which means, it runs every minute)
+
+We highly recommend turning off the logging or at least reducing it to prevent the flow from writing too much data into the history. 
+##### Authenticate and request a Token
+For this specific API, the authentication is a bit more complex compared to basic auth or a simple "request a token" as we might know it from the easy-to-use-api from Directus.
+The general process to get a token which can be used to access actual device data contains three steps:
+
+1. First, we use the regular login credentials (the one I can log in within the mobile App) and pass them to an auth service provided by Google. If successfully logged in, we receive a first token.
+2. With this token we can now secondly authenticate against the actual vendor server. But again we just get a second token.
+3. Within the third request we pass the 2nd token to the vendor API once again. For this request, we will receive a last token that can be used to fetch actual data from the API.
+
+!(Pasted image 20240214224605.png)
+
+As all of these tokens will expire after some time, it could be possible to save the tokens and reuse them instead of chaining all the requests for every run of the flow. For a simple way to setup we run it within a single Flow and pass each token to the next operation. 
+
+To pass data from one Flow Operation to another Operation, the `{{ … }}` syntax can be used. With the `{{ … }}` you can refer to any object that is currently part of the Flow. Here the functionality of named Operation keys becomes handy to easily identify and maintain the Flow.
```

Comment:
Again, remove this. Not your job to teach them how to use Flows. Jsut the specifics about this project